### ISSUE TITLE
Question about the code in function "data_transform"

### ISSUE THREAD

**Issue by seekFire** [yijingru/BBAVectors-Oriented-Object-Detection#69]
I read the code of [these line](https://github.com/yijingru/BBAVectors-Oriented-Object-Detection/blob/12fb6d60bab69c81650fde72291a2c8177c2e338/datasets/base.py#L90-L106) in datasets/base.py, I will use the illustration shown below to describe my question:
![question](https://user-images.githubusercontent.com/26456083/110112542-da1cb380-7dec-11eb-8407-26a9183d85a4.png)

I think the code under the first if statement is used for addressing the case of coordinates overflow like the situation shown in Figure.1.(the x-coord of point **c** is larger than image width). After running the code in line 93~94, the point **c** will be replaced by the new point **c'**, shown in Figure.2. So the ground truth bbox will compose of new four points(a, b, c' and d) rather than (a, b, c, d), which is shown in Figure.3. Under this situation,  the minimum enclosing rectangle of this bbox may either be the same with original bbox(solid green line), or be another coordinates-overflowed box(blue dot line).

**Comment by yijingru** [comment]
Good point. I did this correction because if the coordinates overflow, it would be hard for the network to measure the correct `t, r, b, l` vector when the center is not changed (t!=b, l!=r) or the center position will not be correctly captured. To reduce errors, I made a constraint `iou>0.6`. But it should have a tradeoff.
